File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/javascript/exercises/practice/wordy/wordy.js
export const answer = (question) => {
  // Normalize and validate question format
  const normalized = question.toLowerCase().replace(/[^a-z0-9\s-]/g, '').trim();
  if (!normalized.startsWith('what is ')) {
    throw new Error('Unknown operation');
  }

  const rest = normalized.slice('what is '.length).trim();
  if (!rest) {
    throw new Error('Syntax error');
  }

  const tokens = rest.match(/(-?\d+|plus|minus|multiplied by|divided by)/g) || [];
  if (tokens.join(' ') !== rest) {
    throw new Error('Unknown operation');
  }

  if (tokens.length % 2 === 0 || tokens.length === 0) {
    throw new Error('Syntax error');
  }

  // Validate token sequence
  if (isNaN(tokens[0])) {
    throw new Error('Syntax error');
  }

  for (let i = 1; i < tokens.length; i += 2) {
    if (!['plus', 'minus', 'multiplied by', 'divided by'].includes(tokens[i])) {
      throw new Error('Unknown operation');
    }
    if (i + 1 >= tokens.length || isNaN(tokens[i + 1])) {
      throw new Error('Syntax error');
    }
  }

  let result = parseInt(tokens[0]);
  
  for (let i = 2; i < tokens.length; i += 2) {
    const operation = tokens[i];
    const operand = parseInt(tokens[i+1]);
    
    switch(operation) {
      case 'plus':
        result += operand;
        break;
      case 'minus':
        result -= operand;
        break;
      case 'multiplied by':
        result *= operand;
        break;
      case 'divided by':
        if (operand === 0) throw new Error('Division by zero');
        result = Math.floor(result / operand);
        break;
    }
  }
  
  return result;
};

File: src/python-2025-02-24-deepclaude-rust/python/exercises/practice/wordy/wordy.py
def answer(question):
    if not question.startswith("What is "):
        raise ValueError("unknown operation")
    if not question.endswith("?"):
        raise ValueError("syntax error")
    content = question[8:-1].strip()
    if not content:
        raise ValueError("syntax error")
    tokens = content.split()
    if not tokens:
        raise ValueError("syntax error")
    try:
        result = int(tokens[0])
    except ValueError:
        raise ValueError("syntax error")
    i = 1
    while i < len(tokens):
        if tokens[i] in ('plus', 'minus'):
            op = tokens[i]
            op_len = 1
        elif tokens[i] in ('multiplied', 'divided'):
            if i + 1 >= len(tokens) or tokens[i+1] != 'by':
                raise ValueError("unknown operation")
            op = tokens[i]
            op_len = 2
        else:
            try:
                int(tokens[i])
                raise ValueError("syntax error")
            except ValueError:
                raise ValueError("unknown operation")
        if i + op_len >= len(tokens):
            raise ValueError("syntax error")
        operand_str = tokens[i + op_len]
        try:
            operand = int(operand_str)
        except ValueError:
            raise ValueError("syntax error")
        if op == 'plus':
            result += operand
        elif op == 'minus':
            result -= operand
        elif op == 'multiplied':
            result *= operand
        elif op == 'divided':
            result //= operand
        else:
            raise ValueError("unknown operation")
        i += op_len + 1
    if i != len(tokens):
        raise ValueError("syntax error")
    return result
